packages/rum-core/src/performance-monitoring/performance-monitoring.js

```diff
@@ -186,16 +186,24 @@ class PerformanceMonitoring {
   }
 
   filterTransaction(tr) {
-    var performanceMonitoring = this
-    var transactionDurationThreshold = this._configService.get(
+    const transactionDurationThreshold = this._configService.get(
       'transactionDurationThreshold'
     )
-    var duration = tr.duration()
-    if (
-      !duration ||
-      duration > transactionDurationThreshold ||
-      !tr.spans.length
-    ) {
+    const duration = tr.duration()
+    if (!duration || duration > transactionDurationThreshold) {
+      this._logginService.debug(
+        `Transaction was discarded! transaction duration: ${duration} contains falsy value or it is greater than transactionDurationThreshold: ${transactionDurationThreshold} config`
+      )
+
+      return false
+    }
+
+    if (tr.spans.length === 0) {
+      if (__DEV__) {
```

Comment:
Why only if `__DEV__`?

Reply:
production bundle size will get bigger, so we remove those debug logs and save bytes on our final bundle size. Optimization

Reply:
Yeah I was wondering why here and not elsewhere. I suppose this one is a bit more unusual.

Reply:
But now I see you've added `__DEV__` in the other spot too, so I retract my question ;)